test/unit/alignment/pairwise/policy/unbanded_score_trace_dp_matrix_policy_test.cpp

```diff
@@ -0,0 +1,136 @@
+    using base_t::current_column;
+    using base_t::next_column;
+
+    using base_t::dimension_first_range;
+    using base_t::dimension_second_range;
+    using base_t::score_matrix;
+    using base_t::trace_matrix;
+};
+
+template <typename alloc_types>
+struct unbanded_score_trace_test : public ::testing::Test
+{
+    using score_alloc_t = std::allocator<std::tuple_element_t<0, alloc_types>>;
+    using trace_alloc_t = std::allocator<std::tuple_element_t<1, alloc_types>>;
+
+    auto fixture()
+    {
+        return policy_mock<score_alloc_t, trace_alloc_t>{};
+    }
+};
+
+using test_type = std::tuple<std::tuple<int32_t, int32_t>, seqan3::detail::trace_directions>;
+
+TYPED_TEST_CASE(unbanded_score_trace_test, ::testing::Types<test_type>);
+
+TYPED_TEST(unbanded_score_trace_test, constructor)
+{
+    using mock_t = decltype(this->fixture());
+
+    EXPECT_TRUE(std::is_default_constructible_v<mock_t>);
+    EXPECT_TRUE(std::is_copy_constructible_v<mock_t>);
+    EXPECT_TRUE(std::is_move_constructible_v<mock_t>);
+    EXPECT_TRUE(std::is_copy_assignable_v<mock_t>);
+    EXPECT_TRUE(std::is_move_assignable_v<mock_t>);
+    EXPECT_TRUE(std::is_destructible_v<mock_t>);
+}
+
+TYPED_TEST(unbanded_score_trace_test, allocate_matrix)
+{
+    std::string seq1{"garfieldthecat"};
+    std::string seq2{"garfieldthefatcat"};
+
+    auto mock = this->fixture();
+
+    mock.allocate_matrix(seq1, seq2);
+
+    EXPECT_EQ(mock.dimension_first_range, seq1.size() + 1);
+    EXPECT_EQ(mock.dimension_second_range, seq2.size() + 1);
+    EXPECT_EQ(mock.score_matrix.size(), seq2.size() + 1);
+    EXPECT_EQ(mock.trace_matrix.size(), (seq1.size() + 1) * (seq2.size() + 1));
+}
+
+TYPED_TEST(unbanded_score_trace_test, current_column)
+{
+    std::string seq1{"garfieldthecat"};
+    std::string seq2{"garfieldthefatcat"};
+
+    auto mock = this->fixture();
+
+    mock.allocate_matrix(seq1, seq2);
+
+    auto zip_view = mock.current_column();
+
+    EXPECT_EQ(seqan3::size(zip_view), seq2.size() + 1);
+
+    using value_t = seqan3::value_type_t<decltype(zip_view)>;
+    EXPECT_EQ(std::tuple_size_v<value_t>, 2u);
+}
+
+TYPED_TEST(unbanded_score_trace_test, next_column)
+{
+    std::string seq1{"garfieldthecat"};
+    std::string seq2{"garfieldthefatcat"};
+
+    auto mock = this->fixture();
+
+    mock.allocate_matrix(seq1, seq2);
+
+    // Get the active column
+    auto zip_view = mock.current_column();
+    for (auto && entry : zip_view)
+    {
+        std::get<0>(entry) = std::tuple{10, -10};
+        std::get<1>(entry) = seqan3::detail::trace_directions::diagonal;
+    }
+
+    // Get the same active column again
+    auto zip_view_2 = mock.current_column();
+    for (auto const entry : zip_view_2)
+    {
+        EXPECT_TRUE((std::get<0>(entry) == std::tuple{10, -10}));
+        EXPECT_TRUE((std::get<1>(entry) == seqan3::detail::trace_directions::diagonal));
+    }
+
+    // Go to next active column and check if active column has moved.
```

Comment:
how do you notice if it moved? it has the same score values as zip_view_2?

Reply:
yes it is the expected behavior to have the same score, since it uses only on column and recycles this.

Reply:
The other values have changed though.